apps/google-analytics-4/lambda/src/services/googleApiService.ts

```diff
@@ -44,14 +45,51 @@ export class GoogleApiService {
     return accountSummaries;
   }
 
-  sortReportResponse = (rows: ReportRowType[]) => {
-    rows.filter(
-      (row: ReportRowType) => row.dimensionValues?.length && row.dimensionValues[0].value
+  formatDate = (dateValue: Date) => dateValue.toISOString().slice(0, 10).split('-').join('');
+
+  deriveDate = (dateValue: string) =>
+    new Date(
+      `${dateValue.substring(0, 4)}-${dateValue.substring(4, 6)}-${dateValue.substring(6, 8)}`
     );
 
-    rows.sort((v1: ReportRowType, v2: ReportRowType) => {
-      return Number(v1.dimensionValues[0].value) - Number(v2.dimensionValues[0].value);
+  mapDates = (startDate: string, endDate: string) => {
+    const dates = [];
+    const date = new Date(new Date(startDate).setUTCHours(0, 0, 0, 0));
+    const end = new Date(new Date(endDate).setUTCHours(0, 0, 0, 0));
+
+    while (date <= end) {
+      dates.push(new Date(date));
+      date.setUTCDate(date.getUTCDate() + 1);
+    }
+
+    return dates;
+  };
+
+  supplementDates = (rows: ReportRowType[], _startDate?: string, _endDate?: string) => {
```

Comment:
yeah good point about naming. And the startDate and endDate are just passed in here to then pass into the date `prebuildDateKeyArray` function. But perhaps I can call the function at a different spot to avoid the repetitiveness.